templates/tripleo_myservice/tasks/main.yml.j2

```diff
@@ -19,4 +19,4 @@
 ###############################################################################
 
 - name: Include tasks for deploy stage
-  include_tasks: {{ "{{ deploy_stage }}.yml" }}
+  include_tasks: {{ '"{{' }} deploy_stage {{ '}}' }}.yml"
```

Comment:
I think this is meant to be `include_tasks: "{{ deploy_stage }}.yaml"` but allow for spaces in the deploy_stage?

Does `include_tasks: "\"{{ deploy_stage }}.yaml\""` work?

Reply:
deploy_stage should not have spaces, it should be one of a number of values (host_prep_tasks, upgrade_tasks, external_deploy_tasks, etc), as listed in [1].

The strangeness with the `{{` and the `"` is because the final output will literally be `"{{ deploy_stage }}.yml"` since the entry point for this will always be main.yml, and the deploy_stage var will choose the task to import once the fact setting is done.
See [2] for example.

[1] https://docs.openstack.org/tripleo-docs/latest/developer/tht_walkthrough/service_template_sections.html
[2] https://github.com/infrawatch/tripleo-collectd-ansible-role/blob/master/tasks/main.yml

Reply:
Okay, I think I'll have to play with this myself to truly understand, but if it's working, that's a great starting point! :) Thanks.

Reply:
I re-read this today after having more thoroughly followed the code yesterday and this makes perfect sense now. Thanks! :)

Reply:
Creating a template where the output also has templating mark-up is a bit of a mindfork.